groups/chd-infrastructure/locals.tf

```diff
@@ -50,16 +50,64 @@ locals {
     }
   ] : []
 
-  # Define the NLB FTP target group ARNs (index 0) for ASG registration
-  chd_fe_internal_ftp_target_group_arn = [module.nlb_fe_internal.target_group_arns[0]]
-  chd_fe_external_ftp_target_group_arn = [module.nlb_fe_external.target_group_arns[0]]
-
   # Generate listener configuration for FTP passive ports
   chd_fe_ftp_passive_listeners = [
     for num in range(var.fe_ftp_passive_ports_start, var.fe_ftp_passive_ports_end) : {
       port               = format("%d", num)
       protocol           = "TCP"
-      target_group_index = 0
+    }
+  ]
+
+  # Generate target group configuration for FTP passive ports
+  # Internal NLB TGs
+  chd_fe_internal_ftp_passive_tgs = [
+    for num in range(var.fe_ftp_passive_ports_start, var.fe_ftp_passive_ports_end) : {
+      name                 = "tg-${var.application}-fe-int-ftp-${num}"
+      backend_protocol     = "TCP"
+      backend_port         = num
+      target_type          = "instance"
+      deregistration_delay = 10
+      health_check = {
+        enabled             = true
+        interval            = 30
+        port                = 21
+        healthy_threshold   = 3
+        unhealthy_threshold = 3
+        protocol            = "TCP"
+      }
+      stickiness = {
+        type    = "source_ip"
+        enabled = true
+      }
```

Comment:
We may experience uneven load balancing with sticky sessions given that they are bound to the client source IP rather than based on duration or the presence of a client-side cookie. If clients are behind a NAT device then all connections from that same entity/organisation will be routed to the same target instance. If the source IP is static the same host would be used indefinitely (or until sticky sessions are reset due to target group changes or instance registration/deregistration).

This might be an acceptable tradeoff if the configuration helps to address the complexities of load balancing FTP given its multi-connection nature. Having said that, the backend connections between load balancer and selected target instance should persist for the lifetime of the client connection, so sticky sessions may not provide much benefit in this scenario.

Since the control and data connections are established independently—and associated with different target groups with their own independent sticky sessions—it's likely that the selected target may not be the same, even if that same instance is registered to both target groups. Do we have a mechanism to ensure that the client connects to the same target instance for its data connection as the host that was selected for its control connection?

Reply:
You're correct and, as discussed, since `source_ip` is our only option I have removed the stickiness configuration in [6d754e0](https://github.com/companieshouse/chd-terraform/pull/19/commits/6d754e01aa3d1c05cf84afc5a20ef9ff57c86c6e)